sdk/python/kfp/gcp.py

```diff
@@ -107,4 +109,36 @@ def _set_preemptible(task):
     task.add_node_selector_constraint("cloud.google.com/gke-preemptible", "true")
     return task
 
-  return _set_preemptible
\ No newline at end of file
+  return _set_preemptible
+
+def configure_gcp_connector(
+    op_transformers: List[Callable[[BaseOp], BaseOp]] = [use_gcp_secret()]) -> Callable[[BaseOp], BaseOp]:
+    """Configure GCP settings for GCP connector components.
+    """
+    def _set_gcp_settings(op: BaseOp) -> BaseOp:
+        if op.pod_labels and op.pod_labels['pipelines.kubeflow.org/component-type'] == 'gcp-connector':
```

Comment:
1. `annotations` might be a better section than `labels`
2. What do you think about making the annotation key name explain the behavior. E.g. `needs-pod-name-env: true`?

Reply:
good suggestion, I noticed that kubeflow recently introduced gcp cred webhook to inject gcp credentials. I plan to use the same label to inject for GCP components. Ideally, the pod env should be injected by the same way via webhook. Currently, the webhook is not stable and requires a KF release. 
Adding this pod env transformer in DSL before we adding the webhook in cluster setup.

Reply:
label is better for this use case. Annotations are usually used by storing custom data and not intended for filtering and selector.